### PR TITLE
Modified ApiWithCheckboxInput to account for compatibility layer Apis

### PR DESCRIPTION
**Pull request by Ella-ly**
Modified the onClick property to contain to separate function handlers, `withCallback` and `withPromise`.
`withPromise` will function as it currently does, while `withCallback` will be used by API's that are run in backward compatibility mode.

### CODE REVIEW

**Comment by erinha** [comment]
<p>Analyzed commit id : 66bea6ceb54c72abfa83d9f56618f564fe977861</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 126f401aae255673ebb77e31804e0ee6bb2cb2f5</p>

**Comment by erinha** [comment]
<p>Analyzed commit id : 95dfcfbccf19f8040ac13904b296843d6d87ca98</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 126f401aae255673ebb77e31804e0ee6bb2cb2f5</p>

File: apps/teams-test-app/src/components/utils/ApiWithCheckboxInput.tsx [hunk 1]
```diff
@@ -2,12 +2,18 @@ import * as React from 'react';
...
 
 export interface ApiWithCheckboxInputProps {
   title: string;
   name: string; // system identifiable unique name in context of Teams Client and should contain no spaces
   label: string;
-  onClick: (input: boolean) => Promise<string>;
+  onClick:
+    | ((input: boolean) => Promise<string>)
+    | {
+        withTeamsV2: (input: boolean) => Promise<string>;
```
Comment (MarcinJuraszek) [inline apps/teams-test-app/src/components/utils/ApiWithCheckboxInput.tsx:14]: For the other ones we're using `withPromise` and `withCallback`. We should be consistent.

Comment (Ella-ly) [reply apps/teams-test-app/src/components/utils/ApiWithCheckboxInput.tsx:14]: Thanks. The Apis with checkboxes section are not updating from callbacks to promises, so I would think the `withPromise` and `withCallback` do not explain the scenarios here. If we use `withCallback` just to represent the compatibility layer, that would make sense. I will update them.

Comment (jadahiya-MSFT) [reply apps/teams-test-app/src/components/utils/ApiWithCheckboxInput.tsx:14]: Looking through this, I think we can still follow the `withCallback` and `withPromise` scenarios, the `withPromise` scenario will just replace `onClick.withTeamsV2`

**Comment by erinha** [comment]
<p>Analyzed commit id : 0e2854b999a1acb5834a62334ee9a8c477740945</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: 126f401aae255673ebb77e31804e0ee6bb2cb2f5</p>

**Review by MarcinJuraszek — APPROVED** [review]

**Comment by erinha** [comment]
<p>Analyzed commit id : 030a3e31df071c1d9d5823ede082a29daa61508d</p><hr><details><summary><b><span style="color: green">⯆</span> @microsoft/bundle-analysis-app:</b> -7 Bytes</summary><table><tr><th>Metric Name</th><th>Baseline Size</th><th>Compare Size</th><th>Size Diff</th></tr><tr><td>main.js</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr><tr><td>Total Size</td><td>59.38 KB</td><td>59.37 KB</td><td><span style="color: green">⯆</span> -7 Bytes</td></tr></table></details><hr><p>Baseline commit: e31cb3e70e464c7ecd11a2317fd7c6040acc158c</p>

**Pull request merged**